### DISCUSSION TITLE
Rendering fails with auto imported components [PrimeVue]

### DISCUSSION THREAD

**Discussion by Dunowen** [danielroe/nuxt-vitest#328] [Q&A]
Hey everyone, I'm trying to make my tests work with the nuxt-vitest module and it works fine if I import my PrimeVue components one-by-one in my `<script>` tags. However I'm using a Plugin provided by PrimeVue that auto imports the components that I define in the plugin, so I don't need to manually import them where I want to use them. The plugin looks like this:

```
import { defineNuxtPlugin } from "#app";
import PrimeVue from "primevue/config";

import Button from "primevue/button";

export default defineNuxtPlugin((nuxtApp) => {
	nuxtApp.vueApp.use(PrimeVue, { ripple: true });
	nuxtApp.vueApp.component("Button", Button);
        // .. the rest of the PrimeVue components I want to use with auto import
});
```

My tests throw the following warnings and can't render the PrimeVue components used with auto import:
 ```
[Vue warn]: Failed to resolve component: Button
If this is a native custom element, make sure to exclude it from component resolution via compilerOptions.isCustomElement.
  at <AddContactForm>
  at <ContactList >
  at <Anonymous>
  at <Anonymous ref="VTU_COMPONENT" >
  at <VTUROOT>
```

As a workaround I can manually import the PrimeVue components right in the component files where I want to use them and my test cases run without a problem, but I would rather like to solve this problem with still using the plugin and the auto import feature. Is there a possibility to achieve this? I've tried many different things that didn't work:
- manually running the plugin's code in my test setup file -> I've been warned that the components are defined already, because the plugin's code already ran
- using the `mockComponent()` function to "mock" the PrimeVue components by providing their real implementation (maybe I just didn't do it right, but it didn't work for me, yet I still feel like this could be a solution)

Am I missing something or how could I make the auto imported components render in my test cases? Thank you very much for your help in advance!

**Comment by dpmillerau** [comment]
Hmm, I'm having a similar issue. Did you work out the issue?

**Reply by Dunowen** [reply]
Unfortunately not, I ended up importing the Prime components in my .vue files one-by-one.

**Comment by dpmillerau** [comment]
hmm, I suspect its a bug or at least a missing feature


---
David Miller
IT Specialist
First Aid Officer
WHS Officer
Global Recordings Network Australia
Telling the story of Jesus to every people.
Office: (02) 9899-2211 or +61-2-9899-2211
Direct: 02 8014 6481 ext 120
Mobile: 0416-18-3349   or +61-416-18 3349
***@***.***
www.globalrecordings.net/au
Skype dpmillerau
www.facebook.com/davidpmillerau


________________________________
